src/cross_section.F90

```diff
@@ -1,19 +1,31 @@
 module cross_section
 
   use constants
-  use energy_grid,     only: grid_method, log_spacing
-  use error,           only: fatal_error
+  use energy_grid,      only: grid_method, log_spacing
+  use error,            only: fatal_error
   use global
-  use list_header,     only: ListElemInt
-  use material_header, only: Material
+  use list_header,      only: ListElemInt
+  use material_header,  only: Material
+  use math,             only: faddeeva, broaden_wmp_polynomials
+  use multipole_header, only: FORM_RM, FORM_MLBW, MP_EA, RM_RT, RM_RA, RM_RF, &
+                              MLBW_RT, MLBW_RX, MLBW_RA, MLBW_RF, FIT_T, FIT_A,&
+                              FIT_F, MultipoleArray, max_poly, max_L, max_poles
   use nuclide_header
-  use particle_header, only: Particle
-  use random_lcg,      only: prn, future_prn, prn_set_stream
-  use sab_header,      only: SAlphaBeta
-  use search,          only: binary_search
+  use particle_header,  only: Particle
+  use random_lcg,       only: prn, future_prn, prn_set_stream
+  use sab_header,       only: SAlphaBeta
+  use search,           only: binary_search
 
   implicit none
 
+  ! Allocatable arrays for multipole that are allocated once for speed purposes
+  complex(8), allocatable :: sigT_factor(:)
+  real(8), allocatable :: twophi(:)
+  real(8), allocatable :: broadened_polynomials(:)
```

Comment:
To be clear, the size of the arrays is known at the time of the subroutine invocation, which means they can be fixed-size local arrays that are allocated on the stack. Thus, the allocation is "free" (all it entails is incrementing a stack pointer) and has the benefit that you don't need to make more global, threadprivate variables. It should also exhibit better spatial locality in memory.